cpp/src/io/comp/unsnap.cu

```diff
@@ -280,7 +291,7 @@ __device__ void snappy_decode_symbols(unsnap_state_s *s, uint32_t t)
     if (t == 0) {
       s->q.prefetch_rdpos = cur;
 #pragma unroll(1)  // We don't want unrolling here
-      while (s->q.prefetch_wrpos < min(cur + 5 * batch_size, end)) {}
+      while (s->q.prefetch_wrpos < min(cur + 5 * batch_size, end)) { busy_wait(10); }
```

Comment:
Why 10 in some places and 20 in others?

Reply:
The original times in nanoseconds were 50 or 100. I noticed a significant perf regression when using the same number of cycles (compared to **no** waits). Divided all wait cycles by 5 and got great perf results. Went with this to make the PR available for testing ASAP.
In short, the numbers are not carefully chosen, I'm open to suggestions on the values to use here.